### PR TITLE
Check if group has an active course

### PR DESCRIPTION
**Pull request by balintbartok**
refs: MBL-15548
affects: Student
release note: Fixed groups showing for inactive courses

test plan: see ticket, check normal groups with own account

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/85955774/p/daa3c43834ea3afd0f0e52813f2e14c0">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/85955781/p/8c3dd65f9b47e177b6bbdbaacdc4c920">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/85955727/p/6904e3afc55e5335627ea0276a2692a6">
    
  </a>
</details>

**Comment by inst-danger** [comment]
Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 92.01% | 91.98% | 0.03%

[MBL-15548](https://instructure.atlassian.net/browse/MBL-15548)

#### Affected Apps: Student

#### Release Note: 
Fixed groups showing for inactive courses
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Groups/Group.swift [hunk 1]
```diff
@@ -53,6 +53,10 @@ public final class Group: NSManagedObject, WriteableModel {
         return course
     }
 
+    public var isActive: Bool {
+        courseID == nil || getCourse() != nil
```
Comment (vargaat) [inline Core/Core/Groups/Group.swift:57]: If `getCourse()` returns an instance, would it be possible to check that course's state? If I navigate to `All courses` go back and do a pull-to-refresh the group still appears for a moment.

https://user-images.githubusercontent.com/72396990/128989141-f4c767b6-78cd-4264-bd66-0ad8c161ac29.MP4

**Review by vargaat — CHANGES_REQUESTED** [review]

**Review by davidvarga93 — APPROVED** [review]
QA+1

**Review by vargaat — APPROVED** [review]
QA+1

**Review by idevainstr — APPROVED** [review]
QA+1

**Pull request merged**